### PR TITLE
Add Administration Modal

### PR DESCRIPTION
**Pull request by FinnIckler**
Placement is still TBD, but I added this modal to allow access to administration processes through the new UI. 
Might also Group these a bit better

### CODE REVIEW

File: Frontend/src/ui/Competition.jsx [hunk 1]
```diff
@@ -59,22 +62,114 @@ export default function Competition({ children }) {
...
-                  disabled={!competitionInfo['registration_opened?']}
-                  onClick={(_, data) => {
-                    if (!data.disabled) {
-                      if (competitionInfo.use_wca_registration) {
-                        navigate(`${BASE_ROUTE}/${competitionInfo.id}/register`)
-                      } else {
-                        window.location =
-                          competitionInfo.external_registration_page
+                {canAdminCompetitions ? (
+                  <Modal
```
Comment (kr-matthews) [inline Frontend/src/ui/Competition.jsx:66]: This return is already pretty long, it would be good to have a separate component for the admin options.

**Review by kr-matthews — COMMENTED** [review]
I think the Register button should probably always be present, admin or not.

**Comment by FinnIckler** [comment]
yeah probably, but not sure where else we have space...

**Comment by FinnIckler** [comment]
Closing in favour of using the old sidebar if you are admin

**Pull request closed without merging**